IPython/core/formatters.py

```diff
@@ -775,6 +775,23 @@ class LatexFormatter(BaseFormatter):
     print_method = ObjectName('_repr_latex_')
 
 
+class MathMLFormatter(BaseFormatter):
+    """A MathML formatter.
+
+    To define the callables that compute the MathML representation of your
+    objects, define a :meth:`_repr_mathml_` method or use the :meth:`for_type`
+    or :meth:`for_type_by_name` methods to register functions that handle
+    this.
+
+    The return value of this formatter should be a valid HTML snippet that
+    could be injected into an existing DOM. It should *not* include the
+    ```<html>`` or ```<body>`` tags.
+    """
+    format_type = Unicode('application/mathml+xml')
+
+    print_method = ObjectName('_repr_html_')
```

Comment:
> How does _repr_mathml_ map to the MathML display method?

That's what `print_method` does. It looks for methods with this name, and calls it. Then `format_type` is used to put that in the mathml+xml mime-type field in the format dict.

If it should actually be 'displayed as HTML', then it shouldn't be a mathml mime-type, but the appropriate HTML elements, and actually use `_repr_html_`, and never the mathml mime-type.

I could have sworn that a while ago we added a `_repr_mime_` that would return a mimebundle, so we wouldn't need to declare and register classes and methods for every supported mime-type, but apparently we haven't managed to do that yet. Given the spec, and the DisplayFormatter class, we really don't need dedicated methods and classes for every mime-type, as all arbitrary mime-types are already supported, even those we don't have classes or methods for.